src/clj/athens/self_hosted/event_log.clj

```diff
@@ -6,12 +6,28 @@
     [clojure.edn :as edn]
     [clojure.string :as str]
     [clojure.tools.logging :as log]
-    [fluree.db.api :as fdb]))
+    [fluree.db.api :as fdb])
+  (:import
+    (java.util
+      UUID)))
 
 
 (def ledger "events/log")
 
 
+(defn create-fluree-comp
+  [address]
+  (let [conn-atom    (atom nil)
+        reconnect-fn (fn []
+                       (when-some [conn @conn-atom]
+                         (fdb/close conn))
+                       (reset! conn-atom (fdb/connect address)))
+        comp         {:conn-atom    conn-atom
+                      :reconnect-fn reconnect-fn}]
+    (reconnect-fn)
```

Comment:
It would be better if the creation of the fluree comp did not automatically connect, because it is surprising and it is reasonable to do these two different steps at two different times.

Reply:
I am not sure about this, currently we need to automatically connect on creation of fluree comp in all the cases where this is being used.

 Removing it from here and doing it as a separate step for each case would result in same code in different places then we would have to maintain it all the places.

Reply:
That's a good argument. I'm fine either way then.

I see you've already refactored the fn out. But now reconnect-fn is never called. You need to either:
- call reconnect-fn inside create-fluree-comp
- or call reconnect-fn where create-fluree-comp is called

Up to you which one you prefer.

Reply:
yeah I messed up that part, I first removed it, came up with this arg but did not revert the removal :) Now it's done I went with the first one